inc/agent.class.php

```diff
@@ -306,27 +315,23 @@ public static function showForFleet(PluginFlyvemdmFleet $item) {
 
       $dbUtils = new DbUtils();
 
-      // Total Number of agents
+      // get items
+      $agent = new PluginFlyvemdmAgent();
       $items_id = $item->getField('id');
       $itemFk = $item::getForeignKeyField();
-      $number = $dbUtils->countElementsInTableForMyEntities(
-         static::getTable(),
-         [$itemFk => $items_id]
-      );
+      $condition = "`$itemFk` = '$items_id' " . $dbUtils->getEntitiesRestrictRequest();
+      $rows = $agent->find($condition);
+      $number = count($rows);
 
       // get the pager
       $pager = Html::printAjaxPager(self::getTypeName(1), $start, $number, '', false);
-      $pager = ''; // disabled because the results are not paged yet
-
-      // get items
-      $agent = new static();
-      $condition = "`$itemFk` = '$items_id' " . $dbUtils->getEntitiesRestrictRequest();
-      $rows = $agent->find($condition, '', '');
```

Comment:
We already got the items with the previous request used to count them. The template filters the rows for paging, in a similar way as GLPI does currently.